src/plugins/intel_cpu/src/cpu_map_scheduling.cpp

```diff
@@ -69,4 +69,57 @@ std::vector<std::vector<int>> apply_hyper_threading(bool& input_ht_hint,
     return result_table;
 }
 
+bool check_cpu_pinning(bool cpu_pinning) {
+    bool result_value;
+    auto proc_type_table = get_proc_type_table();
+#if defined(__APPLE__)
+    result_value = false;
+#elif defined(_WIN32)
+    if (proc_type_table.size() == 1) {
+        result_value = cpu_pinning;
+    } else {
+        result_value = false;
+    }
+#else
+    result_value = cpu_pinning;
+#endif
+    return result_value;
+}
+
+bool cpu_pinning_available(const bool cpu_reservation,
```

Comment:
Do we really need to have separate `check_cpu_pinning` and `cpu_pinning_available` functions? The only difference in the logic is handling of Pcore and Ecores. So My question is: cannot we just unite both?
Like we can check that cpuPinning and cpuReservation is available only during generate_stream_info call